### PR TITLE
Split sync item

### PR DESCRIPTION
**Pull request by siscia**
In this PR we are getting rid of the constant `kItemTarFile` (https://github.com/cvmfs/cvmfs/compare/devel...siscia:split_sync_item?expand=1#diff-e9559327380be74c3f80ab56c18c46d2L40)

This implies creating a new class `SyncItemNative` that represents `SyncItem`s that are one the regular filesystem.

This new class is based on the new interface `SyncItem`

### CODE REVIEW

File: cvmfs/sync_union_tarball.cc [hunk 1]
```diff
@@ -176,6 +176,10 @@ void SyncUnionTarball::Traverse() {
           if (filename == ".cvmfscatalog") {
             to_create_catalog_dirs_.insert(parent_path);
           }
+        } else if (sync_entry->IsSymlink() || sync_entry->IsFifo() ||
```
Comment (jblomer) [inline cvmfs/sync_union_tarball.cc]: Isn't this part of another PR?

File: cvmfs/sync_item_tar.cc [hunk 2]
```diff
@@ -104,6 +111,7 @@ catalog::DirectoryEntryBase SyncItemTar::CreateBasicCatalogDirent() const {
   dirent.inode_ = catalog::DirectoryEntry::kInvalidInode;
 
   dirent.linkcount_ = this->tar_stat_.st_nlink;
+  if (dirent.linkcount_ < 1) dirent.linkcount_ = 1;
```
Comment (jblomer) [inline cvmfs/sync_item_tar.cc]: Isn't this part of another PR?

File: test/src/654-tarball_ingest_special_files/main [hunk 3]
```diff
@@ -29,8 +29,10 @@ produce_tarball() {
   dd bs=1024 count=2 2>/dev/null </dev/urandom >tarball_foo/a/b/c/4_change_owner.txt
   chown $CVMFS_TEST_USER tarball_foo/a/b/c/4_change_owner.txt
 
-  ln -s tarball_foo/a/b/another_empty.txt tarball_foo/to_empty_link.txt
-  ln -s tarball_foo/a/b/c/4_change_owner.txt tarball_foo/to_different_owner.txt
+  ln -s empty_file.txt tarball_foo/to_empty_link.txt
+  ln -s a/b/c/4_change_owner.txt tarball_foo/to_different_owner.txt
+
+  mkfifo tarball_foo/fifo
```
Comment (jblomer) [inline test/src/654-tarball_ingest_special_files/main]: This seems to belong to another PR.

**Comment by siscia** [comment]
Superseded by #2146
